### PR TITLE
Reiterated the entire service into proper mvc services

### PR DESCRIPTION
**Pull request by velddev**

### CODE REVIEW

**Review by exsersewo — APPROVED** [review]
lgtm

File: models/error.go [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 // ErrorResponse is the default error
 type ErrorResponse struct {
 	Error string
 }
 
-// New creates a new error response
-func New(reason string) ErrorResponse {
-	return ErrorResponse {
-		Error: reason,
+func NewJSON(reason string) atreugo.JSON {
```
Comment (LovingMelody) [inline models/error.go:11]: I’d like to see error codes here as well in the future.

**Review by LovingMelody — CHANGES_REQUESTED** [review]
I want to see tests before I can approve it. Otherwise, looks good.

**Comment by LovingMelody** [comment]
Approving with the promise of tests in the future before production

**Review by LovingMelody — APPROVED** [review]

**Comment by velddev** [comment]
For now I'll add a todo for tests. The project is too young to properly implement a full test suite in between commits.

One thing I'll work on next is unit tests.

**Pull request merged**